### DISCUSSION TITLE
DB Transaction doesn’t rollback

### DISCUSSION THREAD

**Discussion by adistwn** [adonisjs/core#4417] [Help] [answered]
Hi I used MySQL, and I want using DB Transaction with lucid models, but the rollback function doesn't work properly, this my code :

`const trx = await db.transaction()

const user = new User()
user.useTransaction(trx)
user.username = payload.username
user.name = payload.name
user.email = payload.email
user.identityNumber = payload.identity_number
user.phoneNumber = payload.phone_number
user.password = payload.password
user.roleId = payload.role_id
user.createdBy = userLogged.id
user.updatedBy = userLogged.id
user.save()

await trx.rollback()`

**Comment by holoyan** [comment]
what do you mean by?
>but the rollback function doesn't work properly

what is your expectation? 

@adistwn

**Reply by adistwn** [reply]
I thought when I called 'trx.rollback()' the data will not save to database

**Reply by thetutlage** [reply, marked as answer]
First, you are missing `await` in front of `user.save()` method call.

**Reply by adistwn** [reply]
Thank you
